src/test/ui/async-await/issue-75785-confusing-named-region.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0506]: cannot assign to `*x` because it is borrowed
+  --> $DIR/issue-75785-confusing-named-region.rs:9:5
+   |
+LL | pub async fn async_fn(x: &mut i32) -> (&i32, &i32) {
+   |                          - let's call the lifetime of this reference `'1`
```

Comment:
Now that the highlight points at the argument instead of the return type, this fixes #74072.

This means this PR will change the output of the test I added in #76468, so it will need rebasing on top of that.